src/widgets.ts

```diff
@@ -53,17 +54,38 @@ export class BokehView extends DOMWidgetView {
   private _document: Document | null
   private _receiver: Receiver
   private _blocked: boolean
+  private _msgs: any[]
+  private _idle: boolean
 
   constructor(options?: any) {
     super(options)
     this._document = null
     this._blocked = false
+    this._idle = true
+    this._msgs = []
     const {Receiver} = bk_require("protocol/receiver")
     this._receiver = new Receiver()
     this.model.on("change:render_bundle", () => this.render())
+    if ((window as any).Jupyter != null) {
+      const events = (window as any).require('base/js/events')
+      events.on('kernel_idle.Kernel', () => this._process_msg())
+    } else {
+      (this.model.widget_manager as any).context.sessionContext.statusChanged.connect((_: any, status: any) => {
+        if (status === "idle")
+          this._process_msg()
+      })
+    }
```

Comment:
I really don't love this, currently I have to attach separate kernel busy listeners for classic notebook and JupyterLab. Ideally I'd override the Comm.send callbacks instead.

Reply:
Okay, I've spent way too long on this and can't find a better solution.